lib/ae_mdw/channels.ex

```diff
@@ -49,4 +58,60 @@ defmodule AeMdw.Channels do
     |> Stream.take_while(&match?({^type, txi} when txi < next_txi, &1))
     |> Enum.count()
   end
+
+  defp build_streamer(state, scope, cursor) do
+    fn direction ->
+      Collection.stream(state, @table_activation, direction, scope, cursor)
+    end
+  end
+
+  defp render_list(state, keys),
+    do: Enum.map(keys, fn {_exp, channel_pk} -> render(state, channel_pk) end)
+
+  defp render(state, channel_pk) do
+    Model.channel(
+      initiator: initiator_pk,
+      responder: responder_pk,
+      state_hash: state_hash,
+      amount: amount,
+      updates: [{{last_updated_height, _mbi}, last_updated_txi} | _rest] = updates
+    ) = State.fetch!(state, @table_active, channel_pk)
+
+    %{"hash" => tx_hash, "tx" => %{"type" => tx_type}} = Txs.fetch!(state, last_updated_txi)
+
+    %{
+      channel: Enc.encode(:channel, channel_pk),
+      initiator: Enc.encode(:account_pubkey, initiator_pk),
+      responder: Enc.encode(:account_pubkey, responder_pk),
+      state_hash: :aeser_api_encoder.encode(:state, state_hash),
+      amount: amount,
+      last_updated_height: last_updated_height,
+      last_updated_tx_hash: tx_hash,
+      last_updated_tx_type: tx_type,
+      updates_count: length(updates)
+    }
+  end
+
+  defp deserialize_cursor(nil), do: nil
+
+  defp deserialize_cursor(cursor_bin) do
+    with [exp_height, channel_pk] <-
+           Regex.run(~r/\A(\d+)-(\w+)\z/, cursor_bin, capture: :all_but_first),
+         {:ok, channel_pk} <- Validate.id(channel_pk, [:channel]) do
+      {:ok, {String.to_integer(exp_height), channel_pk}}
+    else
+      _invalid -> ErrInput.Cursor.exception(value: cursor_bin)
```

Comment:
on a `:error` tagged tupple to match https://github.com/aeternity/ae_mdw/blob/874312772484a011e01418195697ae082aed022c/lib/ae_mdw_web/controllers/fallback_controller.ex#L21

Reply:
Good catch, forgot to add it to the function `@spec` and dialyzer didn't catch this error. Fixed now!